domain/src/main/kotlin/no/nav/su/se/bakover/domain/Behandling.kt

```diff
@@ -304,14 +311,18 @@ data class Behandling(
         }
     }
 
-    private open inner class TilAttestering : Behandling.Tilstand {
+    private open inner class TilAttestering : Tilstand {
         override val status: BehandlingsStatus = BehandlingsStatus.TIL_ATTESTERING_INNVILGET
 
         inner class Innvilget : TilAttestering() {
             override fun iverksett(
                 attestant: Attestant,
                 publish: UtbetalingPublisher
             ): Either<UtbetalingPublisher.KunneIkkeSendeUtbetaling, Behandling> {
+                if (attestant.id == this@Behandling.saksbehandletAv?.id) {
```

Comment:
Kunne kanskje sett på om vi skulle generalisert disse feilsituasjonene (KunneIkkeSendeUtbetaling, "KunneIkkeAttester?") til "KunneIkkeIverksette" eller noe lignende, istedenfor å tryne med teknisk feil?

Reply:
Det kan vel argumenteres for at det i dette tilfellet _bør_ tryne fordi koden rundt ikke har guardet godt nok; jeg tenker at autorisasjon bør være ansvarlig for å hindre at vi kommer helt hit.

Reply:
Dette føles som domene-logikk for meg (siden den er avhengig av 2 forskjellige variable) og bør ligge i domenet. Og dette er vel en av de første plassene i denne chainen i domenet? Hvis det er det synes jeg den bør returnere en sealed class med forskjellige objects, og evt data class dersom man trenger context. 

Hvis en tidligere kodesnutt burde hatt guarden, synes jeg ikke vi skal ha guarden her i det hele tatt.